From: John Hubbard <jhubbard@nvidia.com>
Subject: selftests/mm: fix unused variable warning in hugetlb-madvise.c
Date: Thu, 1 Jun 2023 18:33:48 -0700

The dummy variable is required in order to make this work, so declare it
as volatile in order to avoid the clang compiler warning.

Link: https://lkml.kernel.org/r/20230602013358.900637-3-jhubbard@nvidia.com
Signed-off-by: John Hubbard <jhubbard@nvidia.com>
Cc: Mike Kravetz <mike.kravetz@oracle.com>
Cc: Mel Gorman <mgorman@techsingularity.net>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Peter Xu <peterx@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/hugetlb-madvise.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/testing/selftests/mm/hugetlb-madvise.c~selftests-mm-fix-unused-variable-warning-in-hugetlb-madvisec
+++ a/tools/testing/selftests/mm/hugetlb-madvise.c
@@ -65,7 +65,7 @@ void write_fault_pages(void *addr, unsig
 
 void read_fault_pages(void *addr, unsigned long nr_pages)
 {
-	unsigned long dummy = 0;
+	volatile unsigned long dummy = 0;
 	unsigned long i;
 
 	for (i = 0; i < nr_pages; i++)
_
